src/main/java/uk/ac/ebi/atlas/search/GeneSearchDao.java

```diff
@@ -158,13 +166,12 @@ public GeneSearchDao(NamedParameterJdbcTemplate namedParameterJdbcTemplate,
                     Map<Integer, List<Integer>> result = new HashMap<>();
 
                     while (resultSet.next()) {
-                        Integer k = resultSet.getInt("k");
-                        Integer clusterId = resultSet.getInt("cluster_id");
+                        Integer k = Integer.valueOf(resultSet.getString("k"));
```

Comment:
we can use `var` here

Reply:
I cannot use var here. We want to change only DAO class not service and controller layers. Those layers following existing codebase` k` and `cluster_id` as int values.

Reply:
Yes, we can use `var` here it won't affect the return values as you are creating a map of `Map<Integer, List<Integer>>` which is being returned. What I was suggesting is as follows:

`var k = Integer.valueOf(resultSet.getString("k"));`
`var clusterId = Integer.valueOf(resultSet.getString("cluster_id"));`
`var clusterIds = result.getOrDefault(k, new ArrayList<>());`

Reply:
silly me. 👍 . I have interpreted your previous comment differently.

Reply:
fixed it.